### ISSUE TITLE
CodeCov alternatives for future batches

### ISSUE THREAD

**Issue by kxrt** [nus-cs2103-AY2223S1/forum#434]
CodeCov suffers from recurring issues with the GitHub Actions API, as identified in [this 4-day-old post](https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954) by a CodeCov staff member. This has been going on for months now. I'm sure quite a few of the other students have encountered this over the past few tP weeks, as I've heard over discussions. It was never debilitating to our processes but definitely a nuisance now and again.

Some issues from the [CodeCov action repo](https://github.com/codecov/codecov-action/issues):

- codecov/codecov-action#595
- codecov/codecov-action#598 (Main issue with consolidated reports, prompted the staff response.)
- codecov/codecov-action#680
- codecov/codecov-action#754
- codecov/codecov-action#776 

Would it be possible to provide students in future batches with possibly one other coverage tool so that they can work around this issue without decreasing the coverage requirement severity/disabling CodeCov?

**Comment by damithc** [comment]
Thanks for the suggestion @kxrt 
Yes, we will consider this next batch. Most probably advise students to remove CodeCov from CI if it is acting up.

**Comment by parnikkapore** [comment]
Just chiming in here that I thought that coding under coverage enforcement is a goal of the module, i.e. the CodeCov checks should not be removed from the CI (as it is one thing that could fail the workflow and invalidate the PR according to module policies; removing CodeCov would be circumvention under this lens). In the end, we did make it unable to affect the passing of the CI, but it took a while before that felt justified.

**Comment by kxrt** [comment]
@parnikkapore Yep, similar thoughts here. My team would be sitting there re-running checks multiple times until CodeCov passed just because we didn't want to decrease its severity in the workflow.

**Comment by RichDom2185** [comment]
Hi Prof, just chiming in, maybe you could set it so CodeCov still runs but does not fail the CI when it fails, since according to the module requirements, there's no set target on test coverage. Our team did that from this PR: https://github.com/AY2223S1-CS2103T-W16-2/tp/pull/352

**Comment by damithc** [comment]
> Hi Prof, just chiming in, maybe you could set it so CodeCov still runs but does not fail the CI when it fails, since according to the module requirements, there's no set target on test coverage. Our team did that from this PR: [AY2223S1-CS2103T-W16-2/tp#352](https://github.com/AY2223S1-CS2103T-W16-2/tp/pull/352)

Good suggestion @RichDom2185 👍
